modules/phase_field/examples/multiphase/DerivativeMultiPhaseMaterial.i

```diff
@@ -0,0 +1,394 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 40
+  ny = 40
+  nz = 0
+  xmin = -12
+  xmax = 12
+  ymin = -12
+  ymax = 12
+  elem_type = QUAD4
+[]
+
+[GlobalParams]
+  # let's output all material properties for demonstration purposes
+  outputs = exodus
+
+  # prefactor on the penalty function kernels. The higher this value is, the
+  # more rigorously the constraint is enforced
+  penalty = 1e3
+[]
+
+#
+# These AuxVariables hold the directly calculated free energy density in the
+# simulation cell. They are provided for visualization purposes.
+#
+[AuxVariables]
+  [./local_energy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./cross_energy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[AuxKernels]
+  [./local_free_energy]
+    type = TotalFreeEnergy
+    variable = local_energy
+    interfacial_vars = 'c'
+    kappa_names = 'kappa_c'
+    additional_free_energy = cross_energy
+  [../]
+
+  #
+  # Helper kernel to cpompute the gradient contribution from interfaces of order
+  # parameters evolved using the ACMultiInterface kernel
+  #
+  [./cross_terms]
+    type = CrossTermGradientFreeEnergy
+    variable = cross_energy
+    interfacial_vars = 'eta1 eta2 eta3'
+    #
+    # The interface coefficient matrix. This should be symmetrical!
+    #
+    kappa_names = 'kappa11 kappa12 kappa13
```

Comment:
The comments describing an input parameter need to be on the same line as the parameter for them to show up in the right block in peacock